src/GingrichStocking_chart.jl

```diff
@@ -1,132 +1,71 @@
 #Gingrich stocking chart calculated using the original upland oak parameters
 #for more information see:http://oak.snr.missouri.edu/silviculture/tools/gingrich.html
 
-using Plots
-using StatPlots; gr()
-using DataFrames
+using RecipesBase
 
-function gingrich_chart(tpa_in,basal_area_in)
-#define parameters
-a =[-0.0507, 0.1698, 0.0317]
-b =[0.175, 0.205, 0.06]
+amd_convert(qmd) = -0.259 + 0.973qmd
+get_tpa(j, i, a) = (1*j*10)/(a[1]+a[2]*amd_convert(i)+a[3]*i^2)
 
-ba_ticks=collect(0:20:160)
-tpa_ticks=collect(0:50:450)
-
-#QMD lines
-dia = [7, 8, 10, 12, 14, 16, 18, 20, 22]
-#stocking percent horizontal lines
-stk_percent=collect(20:10:110)
-
-#function to convert between artihmetic and quadratic mean
-#need to add this to test?
-function amd_convert(qmd)
-  amd=-0.259+0.973*qmd
+function stklines(stk, dia, a)
+  tpa = get_tpa.(stk', dia, Ref(a))
+  ba = pi * (dia ./ 24).^2 .* tpa
+  tpa, ba
 end
 
-#function to draw stocking lines...needs work
-function make_stklines(stk,dia)
-  BA=Float64[]
-  TPA=Float64[]
-  STK=Int64[]
-  for j in stk
-    for i in dia
-      tpa=(1*j*10)/(a[1]+a[2]*amd_convert(i)+a[3]*i^2)
-      ba=pi*(i/24)^2*tpa
-      push!(TPA,tpa)
-      push!(BA,ba)
-      push!(STK,j)
-    end
+@userplot gingrich_chart
+@recipe function f(dat::gingrich_chart)
+
+  #define parameters
+  a = [-0.0507, 0.1698, 0.0317]
+  b = [0.175, 0.205, 0.06]
+  ba_ticks= 0:20:160
```

Comment:
by the way, now that I'm looking at it, I don't think ba_ticks or tpa_ticks are ever used?